### PR TITLE
[netcore] Remap String ctors to the String:Ctor () methods.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

File: mono/metadata/marshal.c [hunk 1]
```diff
@@ -3543,35 +3545,60 @@ mono_marshal_get_native_wrapper (MonoMethod *method, gboolean check_exceptions,
...
+							break;
+					if (i == csig->param_count) {
+						res = m;
+						break;
+					}
+				}
+			}
+		}
+#else
 		iter = NULL;
```
Comment (jaykrell) [inline mono/metadata/marshal.c]: Move `iter = NULL` to before `#if`?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**